services/user/Tokens/src/RTokens.cpp

```diff
@@ -13,64 +15,229 @@ using namespace UserService;
 using namespace std;
 using namespace psibase;
 
-auto tokenService = QueryableService<Tokens::Tables, Tokens::Events>{Tokens::service};
-
-struct UserBalanceRecord
+auto tokenService = Tokens::Tables{Tokens::service};
+namespace TokenQueryTypes
 {
-   AccountNumber user;
-   uint64_t      balance;
-   uint8_t       precision;
-   TID           token;
-   SID           symbol;
-};
-PSIO_REFLECT(UserBalanceRecord, user, balance, precision, token, symbol);
+   auto realBalance = [](uint64_t bal, const Precision& p) -> uint64_t
```

Comment:
Do we really want to ignore fractional tokens when comparing?

Reply:
It was intentional, but... deleted.